src/main.rs

```diff
@@ -74,15 +75,17 @@ async fn main() -> Result<()> {
     let kakarot_rpc_module = match starknet_provider {
```

Comment:
Could we simplify this to only have to start the service once? Something like return a trait object from the match and then spawn?

Reply:
will be done in another PR